app/controllers/listings_controller.rb

```diff
@@ -10,6 +10,15 @@ def index
   def show
     @host_details = User.find(@listing.user_id)
     @carousel = @listing.images
+    @listing_reviews = Review
+                            .where("bookings.listing_id like #{@listing.id} and reviews.user_id != #{@listing.user_id}")
+                            .joins('JOIN bookings ON reviews.booking_id = bookings.id')
+                            .joins('JOIN users ON users.id = reviews.user_id')
+                            .paginate(page: params[:page], per_page: 3)
+    @review_average = Review
+                            .where("bookings.listing_id like #{@listing.id} and reviews.user_id != #{@listing.user_id}")
+                            .joins('JOIN bookings ON reviews.booking_id = bookings.id')
+                            .pluck('avg(rating)')
```

Comment:
Please revise this and check that there is no longer any error before submitting the PR.
Take not that there is no direct relationship between the Review and Listing.